pkg/acsengine/pki.go

```diff
@@ -133,6 +190,10 @@ func createCertificate(commonName string, caCertificate *x509.Certificate, caPri
 		template.DNSNames = extraFQDNs
 		template.IPAddresses = extraIPs
 		template.ExtKeyUsage = append(template.ExtKeyUsage, x509.ExtKeyUsageServerAuth)
+	} else if commonName == "etcd" {
+		template.IPAddresses = extraIPs
+		template.ExtKeyUsage = append(template.ExtKeyUsage, x509.ExtKeyUsageClientAuth)
```

Comment:
Should server cert support x509.ExtKeyUsageServerAuth only and client cert support x509.ExtKeyUsageClientAuth only? Peer cert maybe support both.

Reply:
For simplicity I just added both for now since one way or the other didn't seem to work but that's on my todo list of small improvements :)